File: Onde-esta-o-Marmore/OndeEstaOMarmore.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
    int n, q, casos = 1;
    cin >> n >> q;
    while(n != 0 && q != 0){
        vector<int> pecas(n);
        for(int i = 0; i < n; ++i) cin >> pecas[i];
        sort(pecas.begin(), pecas.end());
        int consultas[q];
        for(int i = 0; i < q; ++i) cin >> consultas[i];
        cout << "CASE# " << casos++ << ":\n";
        for(int i = 0; i < q; ++i){
            for(int j = 0; j < n; ++j){
                if(pecas[j] == consultas[i]){
                    cout << consultas[i] << " found at " << j + 1 << endl;
                    break;
                }
                else if(j == n - 1 && pecas[j] != consultas[i]) cout << consultas[i] << " not found\n";
            }
        }
        cin >> n >> q;
    }
}

File: Onde-esta-o-Marmore/OndeEstaOMarmore.py
from bisect import bisect_left

n, q = map(int, input().split())
casos = 1
while n != 0 and q != 0:
    pecas = []
    for i in range(n):
        pecas.append(int(input()))
    pecas.sort()
    consultas = []
    for i in range(q):
        consultas.append(int(input()))
    print(f"CASE# {casos}:")
    casos += 1
    for i in range(len(consultas)):
        index = bisect_left(pecas, consultas[i])
        if index != len(pecas):
            print(f"{consultas[i]} found at {index}")
        else:
            print(f"{consultas[i]} not found")
    n, q = map(int, input().split())